### PR TITLE
t: Prevent duplicate test database initialization

### PR DESCRIPTION
**Pull request by okurz**
Our OpenQA::SeleniumTest::call_driver function when called without
arguments would implicitly create a test database. In all cases where
this was used the database was not used at all and just takes time to
initialize and is prone to confuse test writers with two parallel test
database instances.

This commit changes `call_driver` to not initialize its own database
anymore. If potentially necessary one could pass an externally created
database.

Related progress issue: https://progress.opensuse.org/issues/71500

### CODE REVIEW

**Review by Martchus — APPROVED** [review]

File: t/ui/14-dashboard.t [hunk 1]
```diff
@@ -27,7 +27,7 @@ use OpenQA::SeleniumTest;
 
 OpenQA::Test::Case->new->init_data(fixtures_glob => '01-jobs.pl 02-workers.pl 03-users.pl');
 
-my $driver = call_driver();
+my $driver = call_driver;
 unless ($driver) {
     plan skip_all => $OpenQA::SeleniumTest::drivermissing;
```
Comment (kalikiana) [inline t/ui/14-dashboard.t:32]: Might even go one step further and use a one-liner like in `t/ui/13-admin-no-login.t` everywhere

Comment (okurz) [reply t/ui/14-dashboard.t:32]: I will. Expect different follow-up PR

**Review by kalikiana — APPROVED** [review]

**Comment by okurz** [comment]
@kalikiana re-requesting review as I needed to change t/ui/23-audit-log.t which you are also working on and your PR https://github.com/os-autoinst/openQA/pull/3329 is not yet merged. But actually it might help as well because I completely remove the second confusing database.

**Comment by okurz** [comment]
t/05-scheduler-full.t failed with

```
    not ok 1 - Allocated maximum number of jobs that could have been allocated

    #   Failed test 'Allocated maximum number of jobs that could have been allocated'
    #   at t/05-scheduler-full.t line 241.
    #          got: '9'
    #     expected: '10'
    1..1
```

I think I have seen that before but very very seldomly.

**Comment by kalikiana** [comment]
> @kalikiana re-requesting review as I needed to change t/ui/23-audit-log.t which you are also working on and your PR #3329 is not yet merged. But actually it might help as well because I completely remove the second confusing database.

Yeah, when I intended to wrap it up on Friday dropping the second db ended up changing the results again. So it makes sense to take that step here.

However now I'm unsure what those 5 unrelated commits are about - I assume that last one is what this PR was/is about?

**Comment by okurz** [comment]
The fixes are either required for stability or as part of using the single, existing test database. So do you need me to change anything in code or commits or nothing? All of the commits are applied in the order based on dependency so any subset of commits starting with the first one can be merged separately if this is preferred.

**Comment by okurz** [comment]
as you confirmed https://github.com/os-autoinst/openQA/pull/3403/commits/3679cb62d5b68fe79237a469365a1d43907dbb07 looks ok I am going ahead and merge.

**Pull request merged**